### ISSUE TITLE
Incorrect dependency declaration for generateCrashlyticsSymbolFile Gradle task

### ISSUE THREAD

**Issue by guillaume-tgl** [firebase/firebase-android-sdk#5122]
* Android Studio version: Android Studio Flamingo | 2022.2.1 Patch 1
  * Firebase Component: com.google.firebase:firebase-crashlytics-ndk:18.3.7

## Issue

After upgrading to Gradle 8, I get an error when assembling my app in Release mode: 
```
Task ':generateCrashlyticsSymbolFile**' uses this output of task ':merge**NativeLibs' without declaring an explicit or implicit dependency.
```

Here is an excerpt of my build.gradle file:

```
android {
    buildTypes {

        release {
            minifyEnabled true
            proguardFiles getDefaultProguardFile('proguard-android.txt'), '../proguard-project.txt'
            signingConfig signingConfigs.release

            firebaseCrashlytics {
                nativeSymbolUploadEnabled true
            }
        }
    }

    productFlavors {
        pro {
            firebaseCrashlytics {
                unstrippedNativeLibsDir "build/intermediates/merged_native_libs/proRelease/out/lib"
            }
        }
        free {
            firebaseCrashlytics {
                unstrippedNativeLibsDir "build/intermediates/merged_native_libs/freeRelease/out/lib"
            }
        }
    }
}

afterEvaluate {
        bundleFreeRelease.finalizedBy(uploadCrashlyticsSymbolFileFreeRelease)
        bundleProRelease.finalizedBy(uploadCrashlyticsSymbolFileProRelease)
}
```

This looks very similar to #5083. 

## Workaround

I added the following workaround to my build.gradle file to fix the missing task dependency:

```
tasks.all { task ->
    if (task.name.startsWith('generateCrashlyticsSymbolFile')) {
        task.dependsOn 'merge' + (task.name =~ /^generateCrashlyticsSymbolFile(.+)$/)[0][1] + 'NativeLibs'
    }
}
```

**Comment by guillaume-tgl** [comment]
@mrober do you need more details about this issue or is it enough to solve it?

**Comment by egorikftp** [comment]
Still reproduce in 2.9.7

**Comment by egorikftp** [comment]
Can confirm, issue fixed in 2.9.8, thanks 🙂
